platform/ios/src/MGLMapView.mm

```diff
@@ -4729,7 +4736,7 @@ - (void)notifyMapChange:(mbgl::MapChange)change
                             rect = annotationView.frame;
                         }
 
-                        if ( ! CGRectIntersectsRect(rect, self.frame)) {
+                        if ( ! CGRectIntersectsRect(rect, self.frame) && self.deselectWhenOutOfBounds) {
```

Comment:
Given https://github.com/mapbox/mapbox-gl-native/issues/8021#issuecomment-279137067, I’ve changed my mind; if MapKit and Google Maps avoid deselecting the annotation, then that’s what developers expect, and an extra option is unnecessary.

Are there any implications for callout views? Does any of the callout code assume that a callout view (if it exists) is always visible as long as its annotation is selected?